daemon/src/taker_cfd.rs

```diff
@@ -167,6 +185,65 @@ impl<O, M, W> Actor<O, M, W> {
 
         Ok(())
     }
+
+    async fn handle_auto_rollover(&mut self) -> Result<()> {
+        tracing::debug!("Auto rollover");
+
+        let mut conn = self.db.acquire().await?;
+        let cfds = load_all_cfds(&mut conn).await?;
+
+        // cleanup all pending proposals because auto-rollover will create a new one
+        self.current_pending_proposals.clear();
+
+        // Only cfds in state `Open` are allowed for auto-rollover proposal
+        for cfd in cfds
+            .iter()
+            .filter(|cfd| matches!(cfd.state, CfdState::Open { .. }))
+        {
+            let now = OffsetDateTime::now_utc();
+
+            let elapsed_since_creation =
+                now - (cfd.expiry_timestamp() - self.roll_over_params.settlement_interval);
+
+            // Two conditions need to be bet for a roll over to be triggered.
+            // The cfd should not of run longer that the settlement interval.
+            // The cfd should of lived longer than the rollover interval.
+            if self.roll_over_params.settlement_interval > elapsed_since_creation
```

Comment:
I find it confusing if I have to think about the time elapsed since the beginning. 
.
I would rather think about it from the end, i.e. the CFD should always have a remaining expiry of 24h, if it doesn't, roll over.
e.g. 
```
if cfd.expiry_timestamp() - now() <= self.roll_over_params.settlement_interval
  rollover()
```

Reply:
For me this is also easier to follow.

Reply:
ok i will flip it around